### DISCUSSION TITLE
NAT preventing enrollment in Fleet

### DISCUSSION THREAD

**Discussion by echoelric** [Security-Onion-Solutions/securityonion#4619] [Unsupported Versions]
I've had some pretty good success with getting agents on both *NIX and Win hosts, enrolling them in fleet, good to go, but for whatever reason, when I NAT my system using PFSense, the enrollment breaks down.

Functional setup:

- SO Manager on INTERNAL subnet, 10.100.20.X
- Hosts with agents on EXTERNAL subnet 192.168.0.x
- Hosts all default gateway to 192.168.0.254 set up on PFSense
- launcher.flags file pointed at SO manager IP
- PFSense routes traffic to manager via default gateway, all is well, enrollment in Fleet works fine
- Rules in place for allowing traffic from EXTERNAL subnet to INTERNAL functioning as intended

Problem:

- Enabled NAT on PFSense to mask INTERNAL network from EXTERNAL network
- Set flags file to point at gateway on PFSense
- Port forwarding in place to send EXTERNAL traffic over 8090 from gateway to SO manager on INTERNAL network
- TCPDump on both manager and Agent Host confirms 2 way traffic over 8090 is flowing
- No enrollment in fleet

Any thoughts? Can't upload logs here because the issue is at work (no access to github from there) but if there's something I should be looking for in those logs, could be helpful

**Comment by defensivedepth** [comment]
@echoelric My first thought is that it is probably not working because the osquery agent config will only connect to a valid cert that has the hostname / IP of the Fleet server. 

If you are testing on Windows, check out the Application log and look for `Launcher` events - restart the Launcher service and look for connection attempt logs.
